work25/register.js

```diff
@@ -0,0 +1,180 @@
+    formFrag.userName = false
+  }
+
+  submitBtnFrag()
+})
+
+mail.addEventListener('blur', () => {
+  const inputMail = mail.value
+  const result = mailCheck(inputMail)
+  const mailErrorMessage = document.getElementById('mailErrorMessage')
+
+  if (result) {
+    mailErrorMessage && mailErrorMessage.remove()
+    formFrag.mail = true
+  } else {
+    mailErrorMessage || errorMessage(mail)
+    formFrag.mail = false
+  }
+
+  submitBtnFrag()
+})
+
+password.addEventListener('blur', () => {
+  const inputPassword = password.value
+  const result = passwordCheck(inputPassword)
+  const passwordErrorMessage = document.getElementById('passwordErrorMessage')
+
+  if (result) {
+    passwordErrorMessage && passwordErrorMessage.remove()
+    formFrag.password = true
+  } else {
+    passwordErrorMessage || errorMessage(password)
+    formFrag.password = false
+  }
+
+  submitBtnFrag()
+})
+
+/**
+ * 各フォームのバリデーション
+ * ユーザ名： 
+    - 1文字以上、16文字未満
+ * メールアドレス： softbankを参考（https://www.softbank.jp/support/faq/view/10544）
+    - ひとつ目の文字は、アルファベットまたは数字のみ
+    - 文字数は、3文字以上30文字以内
+    - 半角英小文字、数字、「－」（ハイフン）、「．」（ドット）、「＿」（アンダーバー）のみ
+ * パスワード： 
+    - 8文字以上の大小の英数字を交ぜたもの
+ */
+const nameLengthCheck = (inputUserName) => {
+  const minLength = 1
+  const maxLength = 15
+  return inputUserName.length >= minLength && inputUserName.length <= maxLength
+}
+
+const mailCheck = (inputMail) => {
+  const regex = /^[a-z\d][0-9a-z\.-]{2,29}@.+\..{2,}/
+  const result = regex.test(inputMail)
+  return result
+}
+
+const passwordCheck = (inputPassword) => {
+  const regex = /(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])[a-zA-Z0-9]{8,}/
+  const result = regex.test(inputPassword)
+  return result
+}
+
+// エラーメッセージの出力
+const errorMessage = (inputField) => {
+  const errorMessage = document.createElement('span')
+  errorMessage.classList.add('errorMessage')
+
+  switch (inputField) {
+    case userName:
+      errorMessage.id = 'userNameErrorMessage'
+      errorMessage.textContent = '※ユーザ名は1文字以上、15文字以下にしてください'
+      break
+    case mail:
+      errorMessage.id = 'mailErrorMessage'
+      errorMessage.textContent = '※メールアドレスの形式になっていません'
+      break
+    case password:
+      errorMessage.id = 'passwordErrorMessage'
+      errorMessage.textContent = '※8文字以上の大小の英数字を交ぜたものにしてください'
+      break
+  }
+
+  inputField.parentNode.appendChild(errorMessage)
+}
+
+const modalClose = () => {
+  modal.classList.remove('active')
+  mask.classList.remove('active')
+  closeIcon.classList.remove('active')
+}
+
+const modalOpen = () => {
+  modal.classList.add('active')
+  mask.classList.add('active')
+  closeIcon.classList.add('active')
+}
+
+/**
+ *  formFragがすべてtrueか、利用規約を最後まで読んだか を確認する関数
+ *  各フォーム入力時と利用規約を最後まで読んだ時に実行
+ *  全てtrueならsubmitBtnのdisabledを削除、ひとつでもfalseであればdisable付与
+ */
+const submitBtnFrag = () => {
+  if (
+    checkbox.checked === true &&
+    formFrag.userName === true &&
+    formFrag.mail === true &&
+    formFrag.password === true
+  ) {
```

Comment:
ここをもうちょっとスッキリさせたかったのですが、これ以上思いつきませんでした。
良い方法などあればご指導いただけますと嬉しく思います。

Reply:
私は　checkにもフラグを付けて
```suggestion
  const flugs = {
    name: false,
    email: false,
    check: false,
    password: false
};

function checkFlags() {
    const result = Object.values(flugs).every(value => value);
    if (result) {
        submitButton.disabled = false;
    }else{
        submitButton.disabled = true;
    }
}
```

としました！

Reply:
`formFrag`としているところを`frags`に変更して、全て管理すれば良さそうですね！ありがとうございます。
`every`メソッド初めてみました！使い方がわからないのでググります。。。

Reply:
全て同じ`flags`オブジェクトで管理し、` Object.values(flugs).every(value => value);`に修正しました。
初めて` Object.values()`や` every()`を使ったので楽しかったです！ありがとうございます🙇‍♀️
a76cc53